src/equations/1d/compressible_euler.jl

```diff
@@ -69,8 +56,145 @@ function initial_condition_convergence_test(x, t, equation::CompressibleEulerEqu
+
+  du1 = zero(eltype(u))
+  du2 = tmp
+  du3 = tmp
+
+  # Original terms (without performanc enhancements)
+  # du1 = 0
+  # du2 = (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+  #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+  # du3 = (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+  #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+
+  return SVector(du1, du2, du3)
+end
+
+# TODO: Taal remove methods with the signature below
+function source_terms_convergence_test(ut, u, x, element_id, t, n_nodes, equations::CompressibleEulerEquations1D)
+  # Same settings as in `initial_condition`
+  c = 2
+  A = 0.1
+  L = 2
+  f = 1/L
+  ω = 2 * pi * f
+  γ = equations.gamma
+
+  for i in 1:n_nodes
+    x1 = x[1, i, element_id]
+
+    si, co = sincos((t - x1)*ω)
+    tmp = (-((4 * si * A - 4c) + 1) * (γ - 1) * co * A * ω) / 2
+
+    ut[2, i, element_id] += tmp
+    ut[3, i, element_id] += tmp
+
+    # Original terms (without performanc enhancements)
+    # ut[1, i, element_id] += 0
+    # ut[2, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+    # ut[3, i, element_id] += (-(((4 * sin((t - x1) * ω) * A - 4c) + 1)) *
+    #                          (γ - 1) * cos((t - x1) * ω) * A * ω) / 2
+  end
+
+  return nothing
+end
+
+"""
+    boundary_condition_convergence_test(u_inner, orientation, direction, x, t,
+                                        surface_flux_function,
+                                        equations::CompressibleEulerEquations1D)
+
+Boundary conditions used for convergence tests in combination with
+[`initial_condition_convergence_test`](@ref) and [`source_terms_convergence_test`](@ref).
+"""
+function boundary_condition_convergence_test(u_inner, orientation, direction, x, t,
+                                             surface_flux_function,
+                                             equations::CompressibleEulerEquations1D)
+  u_boundary = initial_condition_convergence_test(x, t, equations)
+
+  # Calculate boundary flux
+  if direction == 2 # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
+end
+
+
+# TODO: Taal IC unused?
+"""
+    initial_condition_density_pulse(x, t, equations::CompressibleEulerEquations1D)
+
+A Gaussian pulse in the density with constant velocity and pressure; reduces the
+compressible Euler equations to the linear advection equations.
+"""
+function initial_condition_density_pulse(x, t, equations::CompressibleEulerEquations1D)
+  rho = 1 + exp(-(x[1]^2 ))/2
+  v1 = 1
+  rho_v1 = rho * v1
+  p = 1
+  rho_e = p/(equations.gamma - 1) + 1/2 * rho * v1^2
+  return @SVector [rho, rho_v1, rho_e]
+end
```

Comment:
I also wouldn't remove it. I just added the note `# TODO: Taal IC unused?` since we currently don't test this IC in Taal. I didn't want to modify the test files here, so I just left a note.

Reply:
:+1: